cookies/operations.py

```diff
@@ -474,45 +451,37 @@ def concept_search(text, mock_get):
 
     Parameters
     ----------
-    text : Search text for the URIs
-    mock_get: Mocking the functionality of BlackGoat API
+    query : str
+    This is the search text that will be used by BlackGoat API to query for
+    the URIs.
 
     Returns
     -------
-    concepts: A list of all :class:`.GoatConcept` objects from the search result
+    concepts: list
+    This is a list of all :class:`.GoatConcept` objects obtained from the
+    search result of the BlackGoat API.
     """
 
-    with open('cookies/tests/data/concept_search_results.json', 'r') as f:
-            with open('cookies/tests/data/concept_search_created.json', 'r') as f2:
-                mock_get.return_value = MockSearchResponse(mock_get, f2.read(), f.read(), 200)
-
-    max_calls = 3
     #If no query text is entered, the result from search is None.
-    if not text:
-        print 'Came here'
+    if not query:
         return None
 
+    concepts = None
     #The BlackGoat API for search is used to get a list of all URIs associated
     # with the text entered.
-    concepts = goat.Concept.search(q=text)
-
-    #RuntimeError raised when the calls are not made up to the maximum number
-    # of calls and the status code is 202.
-    if mock_get.call_count != max_calls:
-        raise RuntimeError("Should keep calling if status code 202 is received.")
-
-    #The redirect URL should be followed by the response and the number of
-    # concepts in the result should be 10.
-    args, kwargs =  mock_get.call_args
-    if args[0] != MockSearchResponse.url:
-        raise RuntimeError("Should follow the redirect URL.")
-    if len(concepts) != 10:
-        raise RuntimeError("There should be 10 items in the result set.")
-
-    # All the concepts from the search API should be an instance of
-    # :class:`.GoatConcept`.
-    for concept in concepts:
-        if not isinstance(concept, goat.Concept):
-            raise RuntimeError("Each of which should be a GoatConcept.")
-
-    return concepts
+    try:
+        concepts = goat.Concept.search(q=query)
+    except Exception as e:
+        raise e
+
+    #All the concepts from the search API are iterated to get a dictionary of
+    # lists containing the name, source and uri
+    concept_data = None
+    if concepts:
+        concept_data = []
+        for concept in concepts:
+            concept_data.append({'name':concept.__dict__['data']['name'],
```

Comment:
Why are you using the ``__dict__`` attribute? 

Also, you're missing whitespace after the colon.